addons/hr_timesheet/models/hr_timesheet.py

```diff
@@ -16,6 +16,10 @@ def default_get(self, field_list):
         result = super(AccountAnalyticLine, self).default_get(field_list)
         if 'encoding_uom_id' in field_list:
             result['encoding_uom_id'] = self.env.company.timesheet_encode_uom_id.id
+        employee_id = self._context.get('default_employee_id')
+        if employee_id:
```

Comment:
Cannot we have to check if the `user_id` is already set on the `result`? We erase it? 🤔

Reply:
I agree with @xavierbol. If a user_id is set in the super why overriding it?

Reply:
@xavierbol  @IT-Ideas   can we do this type?

```

employee_id = self._context.get('default_employee_id')
if employee_id:
      employee = self.env['hr.employee'].browse(employee_id)
      if 'user_id' not in field_list or employee.user_id.id  != result.get('user_id'):
           result['user_id'] = employee.user_id.id
```

Reply:
> I was thinking of something like:
> 
> ```python
> employee_id = self._context.get('default_employee_id')
> if not result.get('user_id') and employee_id:
>     employee = self.env['hr.employee'].browse(employee_id)
>     result['user_id'] = employee.user_id.id
> ```
> 
> The only overrides that we would inherit from would be in accounting. So I do not see use cases where the would already be some computing around user_id. That's why I am unsure that we should override a value if one is provided. Don't you agree?

This doesn't work because user_id is set to default so it won't be empty.  [Defaults are set from here](
https://github.com/odoo/odoo/blob/15.0/addons/analytic/models/analytic_account.py#L178-L180)

Reply:
Ok indeed 🙂